guvnor-ala/guvnor-ala-spi/src/main/java/org/guvnor/ala/pipeline/Input.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.pipeline;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.guvnor.ala.config.Config;
+
+/*
+ * Represents the input for a pipeline. Extends HashMap
+ * @see Map, HashMap, Config
+*/
+public class Input
+        extends HashMap<String, String>
+        implements Map<String, String>,
```

Comment:
Good question.. we need to ask @porcelli what kind of magic trickery he was planning there.